probes/requiresLastPushApproval/def.yml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: blocksForcePushOnBranches
+short: Check that the project requires approval of the most recent push.
+motivation: >
+  Requiring approval of the most recent push prevents contributors from sneaking malicious commits into a PR after it has been approved.
```

Comment:
I think there's a lot of overlap here between dismiss stale review and last push approval.  My understanding is the motivation for this particular setting is so that maintainers can't sneak code into PRs not written by themselves.

@laurentsimon ?

Reply:
Discussed in maintainers meeting: Let's wait for @laurentsimon's opinion.